src/Orleans.Runtime/Silo/Watchdog.cs

```diff
@@ -32,37 +33,45 @@ public void Start()
         {
             logger.LogInformation("Starting Silo Watchdog.");
 
-            if (thread is not null)
+            if (gcThread is not null)
             {
                 throw new InvalidOperationException("Watchdog.Start may not be called more than once");
             }
 
             var now = DateTime.UtcNow;
             lastHeartbeat = now;
-            lastWatchdogCheck = now;
             cumulativeGCPauseDuration = GC.GetTotalPauseDuration();
 
-            this.thread = new Thread(this.Run)
+            this.gcThread = new Thread(this.RunGCCheck)
+            {
+                IsBackground = true,
+                Name = "Orleans.Runtime.Watchdog.GCMonitor",
```

Comment:
GCMonitor is a misnomer, since it is monitoring for execution stalls in general. Maybe call it StallMonitor